### ISSUE TITLE
correctly run the sanity checks

### ISSUE THREAD

**Issue by noah1510** [noah1510/update-wrapdb-action#1]
Running the sanity check is the main blocker before tagging the code with v1.
At the moment running the sanity scripts seems to have a few problems that need to be solved:

- [ ] figure out how to run the sanity checks only for the latest commit not for all changes
- [ ] Find the best way to run the sanity checks between the commit and push (using bash or directly from python)
- [ ] Enable the sanity checks by default and fail the action if they fail

The first point is the most problematic.
It is the reason why I set them to be optional.
At the moment  when the sanity checks are run they will run on the whole wrapdb.
In other words they compile every project in the wrapdb.
I don't know why they do this and how to prevent it but it is a problem.
